client/lib/site/utils.js

```diff
@@ -172,5 +172,31 @@ export default {
 
 	isJetpack( site ) {
 		return site && site.jetpack;
+	},
+
+	/**
+	 * Returns the supplied URL without the initial http(s).
+	 * @param  {String}  url The URL to remove http(s) from
+	 * @return {?String}     URL without the initial http(s)
+	 */
+	withoutHttp( url ) {
+		if ( ! url ) {
+			return null;
+		}
+
+		return url.replace( /^https?:\/\//, '' );
+	},
+
+	/**
+	 * Checks whether a site has a custom mapped URL.
+	 * @param  {Object}   site Site object
+	 * @return {?Boolean}      Whether site has custom domain
+	 */
+	hasCustomDomain( site ) {
+		if ( ! site || ! site.domain || ! site.wpcom_url ) {
+			return null;
+		}
+
+		return site.domain !== site.wpcom_url;
```

Comment:
This is OK for now, but should also be ported to https://github.com/Automattic/wp-calypso/blob/master/client/state/sites/selectors.js